### PR TITLE
Add sphinx docs

### PR DESCRIPTION
**Pull request by rlopezcoto**
First stab at adding documentation on lstchain.
Since the repo is growing up, and there are a growing number of users willing to analyze data, this should be helpful to initiate them. It will be complicated to keep it up to date because of the rapid evolution of the software, but we should try to include at least some basic documentation.
At the moment I just included the structure and enabled it in my personal fork:

https://rlopezcoto.github.io/cta-lstchain

if people agree, I'll move to include it in the upstream repo.

### CODE REVIEW

**Comment by mexanick** [comment]
Hi Ruben, thanks for taking care of this. I just have one question: do we need the `index.html` tracked by the repository? I believe it is reproducible by the build from `index.rst` and thus should not be included

**Comment by rlopezcoto** [comment]
@mexanick yes, you are right, I'll correct that and work this PR out a little bit more, so I'll move it to Draft.

**Comment by vuillaut** [comment]
Is this still alive?
Quite needed in my opinion.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,3 +69,36 @@ jobs:
...
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install doc dependencies
+        run: |
+          sudo apt update --yes && sudo apt install --yes git build-essential pandoc graphviz
+          pip install -U pip setuptools wheel
+          pip install -e .[docs]
```
Comment (morcuended) [inline .github/workflows/ci.yml:89]: I think this is currently not working since there is no `docs` requirement in `setup.py`

**Pull request merged**